### ISSUE TITLE
Wish: Package Manager functionalities

### ISSUE THREAD

**Issue by redinkinc** [DynamoDS/Dynamo#4384]
I like to add some additional suggestions to issue #3920 which may be closed since most things seem to work in the current builds?!
- Update packages: The context menu in package manager should have an update functionality. This would be much easier then uninstalling the package and search for them to install them again.
- the search should have an option to show/hide deprecated packages (this is sometimes confusing. e.g. when you are searching for "Clockwork")
- if a depending package is already installed, do not reinstall it (eg when installing dynanimator with clockwork already installed).
- below the library in the "Add" dropdown there should also be the possibility to add local packages/costum nodes. This should allow users to browse a local path or dyf file and add a copy to either C:\Users<USER>\AppData\Roaming\Dynamo\0.8\definitions or \packages and add them to the library. Best would be to also have a checkbox "remove/keep original file".

Thanks, Fabian

**Comment by monikaprabhu** [comment]
@redinkinc , Aboslutely very good requirements- I'm internally tracking them at 
MAGN-7300 - As a user i want to update my packages using Update packages: 
MAGN-7301 - As a user search should have an option to show/hide deprecated packages 
MAGN-7302 - Do not reinstall the package if depending package is already installed 
MAGN-7303 - As a User I want option to load packages/Custom node from local computer into Library UI.
@pboyer

**Comment by elayabharath** [comment]
@redinkinc Would you have some time to take a look at the ideas we have in mind for Package manager? It has all your wishes and more https://docs.google.com/document/d/1GXidznhcULkh0P1h-OrWwQGhpXTvxsx5o1CZNt5Fr9M/edit?usp=sharing

Would be awesome to have your thoughts :)

**Comment by redinkinc** [comment]
Had a look on that document. Think it looks great.
Perhaps you should also adress  #3433 in that file.
Fabian

**Comment by redinkinc** [comment]
Hey guys, how is this new Package Manager going?
I have a few more comments from what I experienced in the last few weeks. I also added them to google docs, but I don't know if they are tracked there. So here we go:
1. It should be possible to delete package, not just only deprecate them. there are many experimental uploads that never will be used.
2. There should be a way of updating package without having the package installed (perhaps at the package manager website). In my case, I'm working with .dlls in c# and it is always a mess with starting stable and unstable builds when packages are installed.
3. Users should be forced to leave a comment when voting for or against a package. I have some votes for my packages without knowing what is wrong or good. This would definitly help developers to improve packages.
4, It should be stated somewhere if a package "only" adds custom nodes to the lybrary using existing functions or if it enhances the Dynamo functionality using .dlls.
anyhow, thanks for the great work you've been doing. Absolutely waiting for the "next big thing" on Dynamo.
